tapgarden/custodian.go

```diff
@@ -125,9 +125,9 @@ type Custodian struct {
 	// statusEventsSubsMtx guards the general status events subscribers map.
 	statusEventsSubsMtx sync.Mutex
 
-	// events is a map of all transaction outpoints and their ongoing
-	// address events of inbound assets.
-	events map[wire.OutPoint]*address.Event
+	// events is a thread safe map of all transaction outpoints and their
+	// ongoing address events of inbound assets.
+	events sync.Map
```

Comment:
Making this thread safe is only required because we want to clean up proofs in the `receiveProof()` method which is a different goroutine than the `watchInboundAssets()` main event loop.
Couldn't we instead just do:
```
c.proofSubscription.NewItemCreated.ChanIn() <- addrProof.Blob
```
at the very end of `receiveProof()` to feed it into the main event loop, removing the need for the `sync.Map` refactor that IMO makes the code a bit less readable with the type casts?

Reply:
That's a great idea! Thanks for taking a look. I've gone with the `c.proofSubscription` channel method.